ambry-router/src/main/java/com.github.ambry.router/GetOperation.java

```diff
@@ -44,8 +44,8 @@
   protected final Callback<T> operationCallback;
   protected final BlobId blobId;
   protected final Time time;
-  protected boolean operationCompleted = false;
-  protected Exception operationException;
+  protected volatile boolean operationCompleted = false;
```

Comment:
Is volatile mandatory? this is updated only during handleResponse() calls if I am not wrong.